R/trans-numeric.r

```diff
@@ -46,6 +46,36 @@ boxcox_trans <- function(p) {
   )
 }
 
+
+#' Box-Cox power transformation.
+#'
+#' @param p Exponent of Box-Cox transformation.
+#' @details The Box-Cox transformation is suitable for strictly positive values.
+#' @references Box, G. E., & Cox, D. R. (1964). An analysis of transformations.
+#' Journal of the Royal Statistical Society. Series B (Methodological), 211-252.
+#' \url{https://www.jstor.org/stable/2984418}
+#'
+#' @export
+boxcox_trans <- function(p) {
+  if (abs(p) < 1e-07) return(log_trans())
+
+  trans <- function(x) {
+    if (any(x < 0)) {
```

Comment:
Perfect, thanks!

Reply:
Can `boxcox_trans` include `offset = 0`as an argument, to allow both versions of the Box-Cox transformation, see #151 ?

The modulus transformation as described by John and Draper assumed `offset = 1`, but in reality it ought to be general. So I suggest adding it as an argument to `modulus_trans` too.

Reply:
That feels slightly too overparameterised to me, given how few people know about the modulus transformation.

Reply:
I agree that few people know of the modulus transformation, including me until a few weeks ago when I saw your mention of it.

But that said, if we are pointing people with negative data towards it, is it not incumbent on us to highlight how it differs from `boxcox?` As well as the sign difference there is also the offset difference, which is far from obvious unless pointed out, and adding an `offset` parameter would do that. Its addition would also make both `boxcox` and `mt` much more flexible as transformations.

If I can't sell you the idea of the offset parameter for `mt`, do please consider it for `boxcox` as it is an important element of Box & Cox (1964).

Reply:
Ok, that sounds reasonable. @dpseidel can you please add to both?